server/src/main/java/org/opensearch/search/aggregations/bucket/terms/InternalSignificantTerms.java

```diff
@@ -209,6 +220,9 @@ protected InternalSignificantTerms(StreamInput in) throws IOException {
         super(in);
         requiredSize = readSize(in);
         minDocCount = in.readVLong();
+        shardSize = BucketUtils.suggestShardSideQueueSize(requiredSize);
+        shardMinDocCount = 0;
+        bucketCountThresholds = new TermsAggregator.BucketCountThresholds(minDocCount, shardMinDocCount, requiredSize, shardSize);
```

Comment:
doesn't look like we are using the `shardSize` and `shardMinDocCount` related parameters here and instead using it from `BucketCountThresholds`. We can construct different type of `BucketCountThresholds` and throw exception from the shard level methods if it is called for finalReduce as that is not expected. For shard level path it can use the current `BucketCountThreshold` class as is. Similar thing can be done for `InternalTerms`